src/reminder/reminder.js

```diff
@@ -0,0 +1,59 @@
+import moment from 'moment'
+
+/**
+ * The Reminder object contains the timerId and metadata of the event
+ */
+class Reminder {
+  // Timer ID
+  timerId = 0
+
+  // Event Metadata
+  title = ''
+  uid = ''
+  start = null
+
+  constructor(timeout, title, uid, start) {
+    /**
+     * Gets called when the app is first loaded and the Reminder object is created.
+     */
+    this.setReminderInterval(timeout)
+
+    // Set Metadata
+    this.title = title
+    this.uid = uid
+    this.start = start
+  }
+
+  setReminderInterval = timeout => {
+    // If duration is positive, enable reminder
+    if (timeout > 0) {
+      // Clear existing timer
+      if (this.timerId) {
+        clearTimeout(this.timerId)
+      }
+
+      this.timerId = setTimeout(this.notifyUser, timeout)
+    }
+  }
+
+  // Popup the notification to remind user about the event
+  notifyUser = () => {
+    if (Notification.permission !== 'granted') {
+      Notification.requestPermission()
+    } else {
+      const notification = new Notification(
+        `${this.title} takes place ${moment.utc().to(moment(this.start))}`,
```

Comment:
This looks wrong, not sure what is wrong

Reply:
@friedger -- It's not wrong. The code for `this.uid` had scope issues. My bad!

I haven't faced the above issue of timezone difference when I updated a Reminder. I'm assuming you've just added a Reminder and was waiting for the notification correct? If so, I need to see what's the value of `event.start` being passed and how's it differing with update scenario. Thanks for pointing this out.

Reply:
The issue could be that we take the event from the text field. The datatime picker does handle timezone already. The datetime field should always contain zulu time.